Final/CharCreator.tsx

```diff
@@ -0,0 +1,744 @@
+import { useState } from "react";
+import { View, Text, ScrollView, TouchableOpacity, TextInput} from "react-native";
+import { Picker } from "@react-native-picker/picker";
+import { RadioButton } from "react-native-paper";
+import { StyleSheet } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { white } from "react-native-paper/lib/typescript/styles/themes/v2/colors";
+
+export function CharCreator() {
+    const navigation = useNavigation();
+
+    const [character, setCharacter] = useState({
+        id: '',
+        name: '',
+        race: '',
+        subrace: '',
+        mclass: '',
+        subclass: '',
+        background: '',
+    });
+
+    const handleNameChange = (name : string) => {
+        setCharacter({ ...character, name });
+    };
+
+    const [selectedOption, setSelectedOption] = useState(null);
+    const [selectedRace, setSelectedRace] = useState(String || null);
+    const [selectedSubrace, setSelectedSubrace] = useState(String || null);
+    
+    const [selectedClass, setSelectedClass] = useState(String || null);
+    const [selectedSubclass, setSelectedSubclass] = useState(String || null);
+
+    const [selectedBackground, setSelectedBackground] = useState(String || null);
+
+    
+    const customizationSections = {
+        race: <Text></Text>,
+        class: <Text></Text>,
+        background: <Text></Text>
+    };
+
+    const handleDropdownChange = (itemValue) => {
+        setCharacter({ ...character, itemValue });
+        setSelectedOption(itemValue);
+    }
+
+    const handleClassChange = (mclass : string) => {
+        setCharacter({ ...character, mclass });
+        setSelectedClass(mclass);
+        setSelectedSubclass('');
+    }
+
+    const handleSubclassChange = (subclass : string) => {
+        setCharacter({ ...character, subclass });
+        setSelectedSubclass(subclass);
+    }
+
+    const handleBackgroundChange = (background : string) => {
+        setCharacter({ ...character, background });
+        setSelectedBackground(background);
+    }
+
+    const handleRaceChange = (race : string) => {
+        setCharacter({ ...character, race });
+        setSelectedRace(race);
+        setSelectedSubrace('');
+    }
+
+    const handleSubraceChange = (subrace : string) => {
+        setCharacter({ ...character, subrace });
+        setSelectedSubrace(subrace);
+    }
+
+    const renderBackgroundSection = () => {
```

Comment:
These repetitive blocks could be programmatically created so they are easier to manage. A quick example would be loop over an array to dynamically build your radio buttons instead of creating each one manually.